src/mqueryfront/src/App.js

```diff
@@ -5,42 +5,51 @@ import QueryPage from "./query/QueryPage";
 import RecentPage from "./recent/RecentPage";
 import StatusPage from "./status/StatusPage";
 import ConfigPage from "./config/ConfigPage";
+import AuthPage from "./auth/AuthPage";
 import axios from "axios";
 import { API_URL } from "./config";
 import "./App.css";
 
+function parseJWT(token) {
+    const base64Url = token.split(".")[1];
+    const base64 = base64Url.replace("-", "+").replace("_", "/");
+    return JSON.parse(Buffer.from(base64, "base64").toString("binary"));
+}
+
 function App() {
     const [config, setConfig] = useState(null);
+    const rawToken = localStorage.getItem("token");
+    const token = rawToken ? parseJWT(localStorage.getItem("token")) : null;
 
     useEffect(() => {
         axios.get(`${API_URL}/server`).then((response) => {
             setConfig(response.data);
         });
     }, []);
 
+    const login = (token) => {
+        localStorage.setItem("token", token);
+        window.location.href = "/";
+    };
+
+    const logout = () => {
+        localStorage.removeItem("token");
+        window.location.href = "/";
+    };
+
     return (
         <div className="App">
-            <Navigation />
+            <Navigation session={token} config={config} logout={logout} />
             <Routes>
-                <Route exact path="/" element={<QueryPage config={config} />} />
-                <Route
-                    path="/query/:hash"
-                    element={<QueryPage config={config} />}
-                />
-                <Route
-                    exact
-                    path="/recent"
-                    element={<RecentPage config={config} />}
-                />
-                <Route
-                    exact
-                    path="/config"
-                    element={<ConfigPage config={config} />}
-                />
+                <Route exact path="/" element={<QueryPage />} />
```

Comment:
I think these routes can be refactored to:
```suggestion
                <Route exact path="/">
                   <QueryPage />
                </Route>
```
But it's a subject for another PR.

Reply:
Are you sure it's possible? I had to redesign this page like tihs after I had to upgrade for the new version of react-router. I don't see any mention of this syntax on the https://reactrouter.com/ page, and when I try to do this I get the following error:

![image](https://user-images.githubusercontent.com/7026881/147274828-86731247-547b-4401-ae1d-31b1fb85efd2.png)

Reply:
Ok, I see now. In v5 they moved to `children` and in v6 they started use `element` property. Eh these JS frameworks, they change so often 😄

So I'm the one that is out of date.

https://github.com/remix-run/react-router/blob/main/docs/upgrading/v5.md